### PR TITLE
Mapillary

### PR DESCRIPTION
**Pull request by RenataMuellerC2C**
GSGMF-1380

### CODE REVIEW

File: examples/mapillarystreetview.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  const dummyToolActivate = new ngeoMiscToolActivate(this, 'dummyActive');
+  ngeoToolActivateMgr.registerTool('mapTools', dummyToolActivate, false);
+}
+
+module.controller('MainController', MainController);
+
+module.constant('ngeoStreetviewOptions', {
+  viewer: 'mapillary',
+  key: 'TFI2dDVCMzRSeEd5MTVvOHVDOG93WDo0MGZkMTlmMThhNjIxODg5',
```
Comment (ger-benjamin) [inline examples/mapillarystreetview.js:106]: @sbrunner I'm not sure we can publish this key in a public repo

File: examples/mapillarystreetview.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  /**
+   * @type {import("ol/style/Style.js").default}
+   */
+  this.style = new olStyleStyle({
+    text: new olStyleText({
+      fill: new olStyleFill({color: '#279B61'}),
+      font: '900 30px "Font Awesome 5 Free"',
+      offsetY: -15,
+      stroke: new olStyleStroke({color: '#ffffff', width: 3}),
+      text: '\uf041',
```
Comment (ger-benjamin) [inline examples/mapillarystreetview.js:64]:
```suggestion
      text: '\uf041', // map-marker
```

File: examples/mapillarystreetview.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2017-2020 Camptocamp SA
```
Comment (ger-benjamin) [inline examples/mapillarystreetview.js:3]:
```suggestion
// Copyright (c) 2020 Camptocamp SA
```

Comment (ger-benjamin) [reply examples/mapillarystreetview.js:3]: that a new file, so....

File: src/streetview/GoogleStreetviewService.js [hunk 4]
```diff
@@ -0,0 +1,143 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2016-2020 Camptocamp SA
```
Comment (ger-benjamin) [inline src/streetview/GoogleStreetviewService.js:3]:
```suggestion
// Copyright (c) 2020 Camptocamp SA
```

File: src/streetview/GoogleStreetviewService.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+   * @param {?import("ol/Map.js").default} map The map
+   * @param {(newCoordinates: import("ol/coordinate.js").Coordinate | null) => void}handlePanoramaPositionChange Position change handler
+   * @param {number} radius The radius
+   * @param {JQuery} $element Element
+   * @ngInject
+   */
+  constructor($scope, map, handlePanoramaPositionChange, radius, $element) {
+    super($scope, map, handlePanoramaPositionChange);
+
+    this.radius_ = radius;
```
Comment (ger-benjamin) [inline src/streetview/GoogleStreetviewService.js:40]: no typedoc ?
(and what's the radius ?)

File: src/streetview/GoogleStreetviewService.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+     * @type {?google.maps.MapsEventListener}
+     * @private
+     */
+    this.panoramaListener_ = null;
+
+    /**
+     * Called when the panorama position changes. Update the location.
+     * @private
+     */
+    this.positionChange_ = () => {
```
Comment (ger-benjamin) [inline src/streetview/GoogleStreetviewService.js:71]: I would like to move this function definition out of the constructor

File: src/streetview/MapillaryService.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2016-2020 Camptocamp SA
```
Comment (ger-benjamin) [inline src/streetview/MapillaryService.js:3]:
```suggestion
// Copyright (c) 2020 Camptocamp SA
```

File: src/streetview/Service.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2016-2020 Camptocamp SA
```
Comment (ger-benjamin) [inline src/streetview/Service.js:3]:
```suggestion
// Copyright (c) 2020 Camptocamp SA
```

File: src/streetview/component.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
+
+<div
+  class="ngeo-streetview-instructions"
+  ng-if="$ctrl.location === null"make
```
Comment (ger-benjamin) [inline src/streetview/component.html:4]: make ?

File: src/streetview/component.js [hunk 10]
```diff
@@ -55,42 +54,56 @@ module.run(
...
+ * The tool has to be declared in the constant 'ngeoStreetviewOptions'. For mapillary
+ * the key (clientId) is needed as well.
+ *
+ * module.constant('ngeoStreetviewOptions', {
+ *  'viewer': 'google',
+ * });
+ *
+ * module.constant('ngeoStreetviewOptions', {
+ * 'viewer': 'mapillary',
+ * 'key': 'TFI2dDVCMzRSeEd5MTVvOHVDOG93WDo0MGZkMTlmMThhNjIxODg5',
```
Comment (ger-benjamin) [inline src/streetview/component.js:84]:
```suggestion
 * 'key': '<your_key>',
```

**Review by ger-benjamin — COMMENTED** [review]
Some first comments.
Good work !

File: examples/mapillarystreetview.js (hunk 1 above)
Comment (sbrunner) [reply examples/mapillarystreetview.js:106]: Effectively it shouldn't ...
Tthe option ngeoStreetviewOptions should be completely removed from here...

File: buildtools/webpack.commons.js [hunk 11]
```diff
@@ -119,7 +119,9 @@ module.exports = function (config) {
   }
 
   const ngeoRule = {
-    test: /\/ngeo\/(?!node_modules\/).*\.js$/,
+    test: (item) => {
```
Comment (ger-benjamin) [inline buildtools/webpack.commons.js:122]: If you revert this change and the next one it works again (at least the apps.)
I don't know exactly why there is this issue but that looked possibly errored.
I've a meeting all this morning, I can't investigate more right now

Comment (ger-benjamin) [reply buildtools/webpack.commons.js:122]: So basically, the path (item) start with /home/.... locally. Never with `ngeo/` or `nodes_modules/`, etc.
I'll try to know what we want to collect, comment it and fix the regexp

**Comment by ger-benjamin** [comment]
@sbrunner @RenataMuellerC2C 
All is green now, and looks working fine :-)

(My second last commit is... A strange thing that had to be done....)

Can you merge and comment your commits @sbrunner ?

**Comment by ger-benjamin** [comment]
The error is that he `can't load 0.js`. In fact that's our `mapillary` webpackChunk defined in src/streetview/MapillaryService.js with:
`import(/* webpackChunkName: "mapillary" */ 'mapillary-js/src/Mapillary').then((Mapillary) => {...}`

If it's not named `vendor~mapillary.js` but `0.js` it's because we remove comments here: https://github.com/camptocamp/ngeo/blob/master/buildtools/webpack.commons.js#L127 
Set `comments` to `true` to preserve the comment `webpackChunkName: "mapillary"` and have the error `can't load vendor~mapillary.js`.

Anyway the error is still there.
**The real problem** is that it uses the path `localhost:3000/contribs/gmf/apps/vendor~mapillary.js` to call our chunk. But the chunk is here: `localhost:3000/vendor~mapillary.js`.

We must set the path used to load this chunk. But I have spent hours to try that and I can't figure out if it's possible like that.

File: buildtools/webpack.commons.js [hunk 12]
```diff
@@ -130,8 +131,11 @@ module.exports = function (config) {
       },
     },
   };
+
+  // Collect every .js file in the node_modules folder except ones that the folder's name
+  // starts with "angular" or "mapillary".
   const otherRule = {
-    test: /\/node_modules\/(?!ngeo\/|angular\/).*\.js$/,
+    test: /\/node_modules\/(?!angular|mapillary\/).*\.js$/,
```
Comment (sbrunner) [inline buildtools/webpack.commons.js:138]:
```suggestion
    test: /\/node_modules\/(?!angular|mapillary)\/.*\.js$/,
```

**Comment by sbrunner** [comment]
The resize of the panel didn't work...

**Comment by ger-benjamin** [comment]
@RenataMuellerC2C The resize of the panel throws an error at the initialization of the app. (look in the debugger console of your browser: `Cannot read property 'resize' of undefined`)
Also, the style "drag" style of the panel is too faded in comparison with the right panel. (I don't know if it's linked to the error, or even with this PR)

**Comment by ger-benjamin** [comment]
I I would have liked to keep the history of commits.... anyway, but please, rename at least the commit named  `f`

**Review by sbrunner — APPROVED** [review]

**Pull request merged**